File: Leetcode/Day-5/Java/Container with most water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0, j = height.length-1, maxi = 0;

        while(i < j){
            maxi = Math.max(maxi, (j-i) * Math.min(height[i], height[j]));
            if (height[i] < height[j]) i += 1;
            else j -= 1;   
        }

        return maxi;
    }
}

File: Leetcode/Day-5/Python/Container with most water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        l = 0
        r = len(height)-1
        maxi = 0
        while l < r:
            maxi = max(maxi, (r - l) * min(height[l], height[r]))
            if height[l] < height[r]:
                l += 1
            else :
                r -= 1
            
        return maxi

        
